types.bal

```diff
@@ -155,40 +147,47 @@ public isolated function mapToExperimentDataResponse(database:ExperimentDataFull
     }
     if (inputFor != ()) {
         dataMapped.inputFor = inputFor;
-        dataMapped.inputForLinks = from var step in inputFor select string `/experiments/${data.experimentId}/timeline/${step}`;
+        dataMapped.inputForLinks = from var step in inputFor
```

Comment:
We will also need to refactor how the inputFor attribute of experiment Data is handled with substeps. My initial guess would be to handle substeps as the step they are part of when coming from data. A file used in a substep would then show that it was used in the step of that substep. But if we view the step then file inputs are broken down into substeps.

Reply:
That means that:
- get experiments/[int experimentId]/data/[string name](string? 'version) shows only the step (not the substep) in which the data was used => use of inputFor would remain as it is.

- get experiments/[int experimentId]/timeline/[int timelineStep]() must be refactored such that in the case of substeps TimelineStepResponse includes some kind of mapping that maps inputData to the respective substep it was used in. 

Do I correctly understand your suggestion?